midend/flattenHeaders.h

```diff
@@ -177,6 +178,7 @@ class FlattenHeaders final : public PassManager {
         passes.push_back(new FindHeaderTypesToReplace(sm));
         passes.push_back(new ReplaceHeaders(sm));
         passes.push_back(new ClearTypeMap(typeMap));
+        passes.push_back(new TypeChecking(refMap, typeMap, true));
```

Comment:
In general you should not end with type checking; if the next pass needs it, it should call it itself.

Reply:
fixed.